c/secp256k1_keccak256_lock_eth.h

```diff
@@ -0,0 +1,405 @@
+#include "bech32.h"
```

Comment:
Diff with c/secp256k1_keccak256_lock.h in master:

```diff
--- master-secp256k1_keccak256_lock.h	2020-09-21 18:43:06.000000000 +0800
+++ secp256k1_keccak256_lock_eth.h	2020-09-21 18:29:24.000000000 +0800
@@ -1,8 +1,3 @@
-#include "ckb_syscalls.h"
-#include "common.h"
-#include "protocol.h"
-#include "secp256k1_helper.h"
-#include "keccak256.h"
 #include "bech32.h"
 
 #define BLAKE2B_BLOCK_SIZE 32
@@ -15,7 +10,9 @@
 #define SCRIPT_SIZE 32768
 #define SIGNATURE_SIZE 65
 
-#define CKB_ADDRESS_PREFIX "ckt"
+#define CKB_ADDRESS_PREFIX "ckb"
+
+#define ENABLE_EIP712 false
 
 #define MAX_OUTPUT_LENGTH 64
 
@@ -361,60 +358,6 @@
 }
 
 /*
- * Verify secp256k1 signature, check the kecack256 hash of pubkey recovered from signature 
- * is equal with the lock script arg.
- * 
- * Arguments:
- * message,  the calculated hash of ckb transaciton.
- * lock_bytes, the signature from transcation witness.
- * lock_args, the args of lock script.
- * 
- * 
- */
-int verify_signature(unsigned char *message, unsigned char *lock_bytes, const void * lock_args){
-
-  unsigned char temp[TEMP_SIZE];
-
-  /* Load signature */
-  secp256k1_context context;
-  uint8_t secp_data[CKB_SECP256K1_DATA_SIZE];
-  int ret = ckb_secp256k1_custom_verify_only_initialize(&context, secp_data);
-  if (ret != 0) {
-    return ret;
-  }
-
-  secp256k1_ecdsa_recoverable_signature signature;
-  if (secp256k1_ecdsa_recoverable_signature_parse_compact(
-          &context, &signature, lock_bytes, lock_bytes[RECID_INDEX]) == 0) {
-    return ERROR_SECP_PARSE_SIGNATURE;
-  }
-
-  /* Recover pubkey */
-  secp256k1_pubkey pubkey;
-  if (secp256k1_ecdsa_recover(&context, &pubkey, &signature, message) != 1) {
-    return ERROR_SECP_RECOVER_PUBKEY;
-  }
-
-  /* Check pubkey hash */
-  size_t pubkey_size = PUBKEY_SIZE;
-  if (secp256k1_ec_pubkey_serialize(&context, temp, &pubkey_size, &pubkey,
-                                    SECP256K1_EC_UNCOMPRESSED) != 1) {
-    return ERROR_SECP_SERIALIZE_PUBKEY;
-  }
-
-  SHA3_CTX sha3_ctx;
-  keccak_init(&sha3_ctx);
-  keccak_update(&sha3_ctx, &temp[1], pubkey_size - 1);
-  keccak_final(&sha3_ctx, temp);
-
-  if (memcmp(lock_args, &temp[12], BLAKE160_SIZE) != 0) {
-    return ERROR_PUBKEY_BLAKE160_HASH;
-  }
-
-  return CKB_SUCCESS;
-}
-
-/*
  * Verify the transaction using secp256k1 as sig algorithm and keccak256 as hash algorithm.
  * 
  * Since not all ethereum wallets support EIP712, the verification will support two hashes 
@@ -428,98 +371,11 @@
  * Witness:
  * WitnessArgs with a signature in lock field used to present ownership.
  */
-int verify_secp256k1_keccak_sighash_all(unsigned char eth_address[BLAKE160_SIZE]) {
+int verify_secp256k1_keccak_eth_sighash_all(unsigned char* message, unsigned char* eth_address, unsigned char* lock_bytes) {
   int ret;
-  uint64_t len = 0;
-  unsigned char temp[TEMP_SIZE];
-  unsigned char lock_bytes[SIGNATURE_SIZE];
-
-  /* Load witness of first input */
-  uint64_t witness_len = MAX_WITNESS_SIZE;
-  ret = ckb_load_witness(temp, &witness_len, 0, 0, CKB_SOURCE_GROUP_INPUT);
-  if (ret != CKB_SUCCESS) {
-    return ERROR_SYSCALL;
-  }
-
-  if (witness_len > MAX_WITNESS_SIZE) {
-    return ERROR_WITNESS_SIZE;
-  }
-
-  /* load signature */
-  mol_seg_t lock_bytes_seg;
-  ret = extract_witness_lock(temp, witness_len, &lock_bytes_seg);
-  if (ret != 0) {
-    return ERROR_ENCODING;
-  }
-
-  if (lock_bytes_seg.size != SIGNATURE_SIZE) {
-    return ERROR_ARGUMENTS_LEN;
-  }
-  memcpy(lock_bytes, lock_bytes_seg.ptr, lock_bytes_seg.size);
 
-  /* Load tx hash */
-  unsigned char tx_hash[BLAKE2B_BLOCK_SIZE];
-  len = BLAKE2B_BLOCK_SIZE;
-  ret = ckb_load_tx_hash(tx_hash, &len, 0);
-  if (ret != CKB_SUCCESS) {
-    return ret;
-  }
-  if (len != BLAKE2B_BLOCK_SIZE) {
-    return ERROR_SYSCALL;
-  }
-
-  /* Prepare sign message */
-  unsigned char message[BLAKE2B_BLOCK_SIZE];
   SHA3_CTX sha3_ctx;
   keccak_init(&sha3_ctx);
-  keccak_update(&sha3_ctx, tx_hash, BLAKE2B_BLOCK_SIZE);
-
-
-  /* Clear lock field to zero, then digest the first witness */
-  memset((void *)lock_bytes_seg.ptr, 0, lock_bytes_seg.size);
-  keccak_update(&sha3_ctx, (unsigned char *)&witness_len, sizeof(uint64_t));
-  keccak_update(&sha3_ctx, temp, witness_len);
-
-  /* Digest same group witnesses */
-  size_t i = 1;
-  while (1) {
-    len = MAX_WITNESS_SIZE;
-    ret = ckb_load_witness(temp, &len, 0, i, CKB_SOURCE_GROUP_INPUT);
-    if (ret == CKB_INDEX_OUT_OF_BOUND) {
-      break;
-    }
-    if (ret != CKB_SUCCESS) {
-      return ERROR_SYSCALL;
-    }
-    if (len > MAX_WITNESS_SIZE) {
-      return ERROR_WITNESS_SIZE;
-    }
-    keccak_update(&sha3_ctx, (unsigned char *)&len, sizeof(uint64_t));
-    keccak_update(&sha3_ctx, temp, len);
-    i += 1;
-  }
-  /* Digest witnesses that not covered by inputs */
-  i = calculate_inputs_len();
-  while (1) {
-    len = MAX_WITNESS_SIZE;
-    ret = ckb_load_witness(temp, &len, 0, i, CKB_SOURCE_INPUT);
-    if (ret == CKB_INDEX_OUT_OF_BOUND) {
-      break;
-    }
-    if (ret != CKB_SUCCESS) {
-      return ERROR_SYSCALL;
-    }
-    if (len > MAX_WITNESS_SIZE) {
-      return ERROR_WITNESS_SIZE;
-    }
-    keccak_update(&sha3_ctx, (unsigned char *)&len, sizeof(uint64_t));
-    keccak_update(&sha3_ctx, temp, len);
-
-    i += 1;
-  }
-  keccak_final(&sha3_ctx, message);
-
-  keccak_init(&sha3_ctx);
   /* personal hash, ethereum prefix  \u0019Ethereum Signed Message:\n32  */
   unsigned char eth_prefix[28]= {
 0x19, 0x45, 0x74, 0x68, 0x65, 0x72, 0x65, 0x75, 0x6d, 0x20, 0x53, 0x69 ,0x67, 0x6e, 0x65, 0x64, 0x20, 0x4d, 0x65, 0x73, 0x73, 0x61, 0x67, 0x65, 0x3a, 0x0a, 0x33, 0x32
@@ -534,12 +390,16 @@
     return CKB_SUCCESS;
   }
 
-  /* Calculate Typed Data hash */
-  ret = calculate_typed_data(message, message);
-  if(ret != CKB_SUCCESS){
-    return ret;
+  if(ENABLE_EIP712){
+    /* Calculate Typed Data hash */
+    ret = calculate_typed_data(message, message);
+    if(ret != CKB_SUCCESS){
+      return ret;
+    }
+
+    /* verify signature with typed data hash */
+    ret = verify_signature(message, lock_bytes, eth_address);
   }
 
-  /* verify signature with typed data hash */
-  return verify_signature(message, lock_bytes, eth_address);
+  return ret;
 }
```

Reply:
Most of the removed codes are now in `secp256k1_keccak256_helper.h`